WordPress/gutenberg
# Table Block: Add a deprecation for the figcaption element class name

Part of #45153

## What?
This PR fixes a problem with broken the table block when updating from WordPress 6.0 to 6.1. Adds a deprecation for the Table block figcaption element class name. 

## Why?
The `wp-element-caption` class has been added to the `figcaption` element by #41239. So if the table has caption text, the block may be broken by updating to 6.1.

If the table does not have a color, the existing migration will not break the block. However, if some style is added by block support, the block should be broken because it does not meet the conditions of the existing migration.

## How?
I have split the existing depreciation into v1 and v2 and added a new v3. The newly added v3 is the definition before #41239 was done.

I also believe that the fixture is not regenerated because table blocks don't have captions by default.

## Testing Instructions

Below is a markup variation of the table created in WordPress6.0. Paste into WordPress 6.1 content with the Gutenberg plugin enabled, and confirm that the block is broken or causes unintended behavior. The actual behavior can be seen in [the video of this comment](https://github.com/WordPress/gutenberg/issues/45153#issuecomment-1285517581).

Then, confirm that this PR won't break the blocks and the expected results are obtained in all variations.

### With text color and caption text

The block will be broken.

<details>
<summary>HTML</summary>

```html

<figure class="wp-block-table"><table class="has-vivid-red-color has-text-color"><tbody><tr><td></td><td></td></tr><tr><td></td><td></td></tr></tbody></table><figcaption>Caption</figcaption></figure>

```
</details>

### With custom text color and caption text

The block will be broken.

<details>
<summary>HTML</summary>

```html

<figure class="wp-block-table"><table class="has-text-color" style="color:#734e4e"><tbody><tr><td></td><td></td></tr><tr><td></td><td></td></tr></tbody></table><figcaption>Caption</figcaption></figure>

```
</details>

### With background color and caption text

The block will lose background color.

<details>
<summary>HTML</summary>

```html

<figure class="wp-block-table"><table class="has-secondary-background-color has-background"><tbody><tr><td></td><td></td></tr><tr><td></td><td></td></tr></tbody></table><figcaption>Caption</figcaption></figure>

```
</details>

### With custom background color and caption text

The block will be broken.

<details>
<summary>HTML</summary>

```html

<figure class="wp-block-table"><table class="has-background" style="background-color:#a16329"><tbody><tr><td></td><td></td></tr><tr><td></td><td></td></tr></tbody></table><figcaption>Caption</figcaption></figure>

```
</details>
